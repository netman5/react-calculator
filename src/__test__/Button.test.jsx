import TestRenderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import Button from '../Components/Buttons';

describe('Use Jest snapshots to test the button component.', () => {
  it('Button component renders properly', () => {
    const props = {
      value: '+',
      label: '+',
      class: 'Button',
    };
    const handleClick = () => {};

    const component = TestRenderer.create(
      <Button
        value={props.value}
        className={props.Button}
        label={props.label}
        handleClick={handleClick}
      />
    );
    expect(component).toMatchSnapshot();
  });
});

describe('Test if button is clicked', () => {
  it('Test if button is clicked once', () => {
    const props = {
      value: '+',
      label: '+',
      class: 'Button',
    };

    const handleClick = jest.fn();

    render(
      <Button
        value={props.value}
        className={props.Button}
        label={props.label}
        handleClick={handleClick}
      />
    );
    userEvent.click(screen.getByText('+'));
    expect(handleClick).toHaveBeenCalledTimes(1);
  });
});

describe('Test if button is clicked', () => {
  it('Test if button is clicked is AC', () => {
    const props = {
      value: 'AC',
      label: 'AC',
      class: 'Button',
    };

    const handleClick = jest.fn();

    render(
      <Button
        value={props.value}
        className={props.Button}
        label={props.label}
        handleClick={handleClick}
      />
    );
    userEvent.click(screen.getByRole('button'));
    expect(handleClick).toHaveBeenCalledTimes(1);
  });
});
