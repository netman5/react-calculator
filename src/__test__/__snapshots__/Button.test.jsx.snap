// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Use Jest snapshots to test the button component. Button component renders properly 1`] = `
<button
  className="Button"
  onClick={[Function]}
  type="button"
  value="+"
>
  +
</button>
`;
